examples/Pin_control/Pin_control.ino

```diff
@@ -53,8 +53,17 @@ void setup()
 {
   Wire.begin();
   Serial.begin(115200);
+
+#ifdef SerialUSB
+  // native USB needs to wait
+  unsigned int begin_time = millis();
+  while (! Serial && millis() - begin_time < 1000) {
+    delay(10);  // but at most 1 sec if not plugged in to usb
+  }
+#endif
+
```

Comment:
Where is SerialUSB coming from? Please provide a source that documents why this is important.

I'd prefer to keep the examples as simple as possible so that their didactic purpose is not compromised. I don't see this in other libraries' example sketches.

Also it makes no sense to have that snippet only in one example sketch and not the others.

Reply:
`SerialUSB` is not well documented, but has become a de facto standard for all Arduinos with "native usb" (mostly because everyone uses the original Zero code to start with). Here it is for the mkrwifi1010, but is identical almost everywhere:
``` 
.arduino15/packages/arduino/hardware/samd/1.8.13/variants/mkrwifi1010/variant.h:#define Serial                      SerialUSB
```

Arduinos with native-usb must wait for the serial connection to finish opening. This is documented:

https://www.arduino.cc/reference/en/language/functions/communication/serial/ifserial/

Note the short reference to "SerialUSB" in the above.

The `#ifdef` block is an attempt to make the code behave properly on the SAMD's without impacting the performance of non-"native usb" systems. Without the `if(Serial)` you will lose up to 3 seconds of Serial output from the Arduino. But, if the Arduino is not connected to a USB host, `if(Serial)` will never become true, and the program will hang. Thus the timeout.

As I wrote this, I realized that the `#ifdef` is not actually necessary. On non native-usb, `if(Serial)` is immediately true, so no delay will happen. Impact on code-size is negligible.

The `if(! Serial)` is becoming more common in examples as the SAMD based Arduinos (and other native-usb chips) become more popular, it is usually expressed as:

```
while (! Serial);
```

e.g.

```
Arduino/libraries/SD/examples/Files/Files.ino:  while (!Serial) {
```

I found 148 uses in library examples (74 libraries), and 23 uses in board examples 116 boards).

Note that the basic formulation will lock up a native-usb Arduino unless it is plugged into a USB Host (desktop/laptop). Thus the timeout.

I'll remove this block from the example, but leave it in the firmware.

Reply:
Fair by me.

I've seen the `while (! Serial);` solution, and would have no problems adding it to all examples due to it's brevity, even if, as you've shown, it is not a 100% fool proof solution.

I'll make a note to add it in my next revision, or if you prefer you could do it within this PR.